lib/spack/spack/solver/asp.py

```diff
@@ -2068,9 +2238,10 @@ def build_specs(self, function_tuples):
             assert action and callable(action)
 
             # ignore predicates on virtual packages, as they're used for
-            # solving but don't construct anything
+            # solving but don't construct anything. Do not ignore error
+            # predicates on virtual packages.
             pkg = args[0]
-            if spack.repo.path.is_virtual(pkg):
+            if spack.repo.path.is_virtual(pkg) and not name.startswith('error_'):
```

Comment:
reverse the conditions here, as checking for the `error_` prefix is way less expensive.